plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -115,6 +115,8 @@
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.net.URI;
```

Comment:
should "Return $path without URL encoding in Iceberg" commit have any test changes/additions?

Reply:
Ideally it should, but let me handle in #13457

Reply:
url encoding (or lack of) should be exercisable independently from double slashes.
eg path containing `%`.

Reply:
Not sure if I understood your suggestion correctly, but just adding `%` wouldn't work because it doesn't pass in `!path.equals(hadoopPath.toString())` in `hadoopPath()`.